src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,164 @@
+package duke;
+
+import exception.*;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Locale;
+
+/**
+ * Handles duke operations and parses input commands.
+ */
+public class Duke {
+    /**
+     * DukeResponse object to handle responses.
+     */
+    private final DukeResponse dukeResponse;
+
+    /**
+     * DukeStorage object to handle history files.
+     */
+    private final DukeStorage dukeStorage;
+    /**
+     * DukeTaskList to handle task list operations.
+     */
+    private final DukeTaskList dukeTaskList;
+
+    /**
+     * Datetime parsed format.
+     */
+    public static DateTimeFormatter format = DateTimeFormatter
+            .ofPattern("dd/MM/yyyy, hh:mma", Locale.US);
+
+    /**
+     * Instantiates a new duke.
+     *
+     * @throws IOException the io exception
+     */
+    public Duke() throws IOException, DukeException {
+        this.dukeStorage = new DukeStorage();
+        this.dukeTaskList = new DukeTaskList(this.dukeStorage.load());
+        this.dukeResponse = new DukeResponse();
+    }
+
+    /**
+     * Parses given input and sets a new current message based on input.
+     *
+     * @param input the input message.
+     * @return integer. 0 represents an terminating entry, 1 represents a continuing entry.
+     * @throws DukeException the duke exception
+     * @throws IOException   the io exception
+     */
+    public int parse(String input) throws DukeException, IOException {
```

Comment:
This method seems a bit long. Perhaps you can use helper methods to make the code shorter?